crates/verify/src/bytecode.rs

```diff
@@ -0,0 +1,506 @@
+    }
+
+    fn data(
+        &self,
+    ) -> Result<figment::value::Map<figment::Profile, figment::value::Dict>, figment::Error> {
+        let mut dict = figment::value::Dict::new();
+        if let Some(block) = &self.block {
+            dict.insert("block".into(), figment::value::Value::serialize(block)?);
+        }
+        if let Some(rpc_url) = &self.rpc_url {
+            dict.insert("eth_rpc_url".into(), rpc_url.to_string().into());
+        }
+        dict.insert("verification_type".into(), self.verification_type.to_string().into());
+
+        // if let Some(root) = self.root.as_ref() {
+        //     dict.insert("root".to_string(), figment::value::Value::serialize(root)?);
+        // }
+        Ok(figment::value::Map::from([(Config::selected_profile(), dict)]))
+    }
+}
+
+impl VerifyBytecodeArgs {
+    /// Run the `verify-bytecode` command to verify the bytecode onchain against the locally built
+    /// bytecode.
+    pub async fn run(mut self) -> Result<()> {
+        // Setup
+        let config = self.load_config_emit_warnings();
+        let provider = ProviderBuilder::new(&config.get_rpc_url_or_localhost_http()?).build()?;
+
+        let code = provider.get_code_at(self.address, None).await?;
+        if code.is_empty() {
+            eyre::bail!("No bytecode found at address {}", self.address);
+        }
+
+        println!(
+            "Verifying bytecode for contract {} at address {}",
+            Paint::green(self.contract.name.clone()),
+            Paint::green(self.address.to_string())
+        );
+        // If chain is not set, we try to get it from the RPC
+        // If RPC is not set, the default chain is used
+        let chain = match config.get_rpc_url() {
+            Some(_) => {
+                let chain_id = provider.get_chain_id().await?;
+                // Convert to u64
+                Chain::from(chain_id.to::<u64>())
+            }
+            None => config.chain.unwrap_or_default(),
+        };
+
+        // Set Etherscan options
+        self.etherscan_opts.chain = Some(chain);
+        self.etherscan_opts.key =
+            config.get_etherscan_config_with_chain(Some(chain))?.map(|c| c.key);
+        // Create etherscan client
+        let etherscan = Client::new(chain, self.etherscan_opts.key.clone().unwrap())?;
+
+        // Get the constructor args using `source_code` endpoint
+        let source_code = etherscan.contract_source_code(self.address).await?;
+
+        let name = source_code.items.first().map(|item| item.contract_name.to_owned());
+        if name.as_ref() != Some(&self.contract.name) {
+            eyre::bail!("Contract name mismatch");
+        }
+
+        let constructor_args = match source_code.items.first() {
+            Some(item) => item.constructor_arguments.clone(),
+            None => {
+                eyre::bail!("No source code found for contract at address {}", self.address);
+            }
+        };
```

Comment:
same, please no matches for Options, if let Some or `let Some() else {bail!}` is preferred here